kernel/register.cc

```diff
@@ -668,8 +688,9 @@ void Backend::extra_args(std::ostream *&f, std::string &filename, std::vector<st
 		if (f != NULL)
 			cmd_error(args, argidx, "Extra filename argument in direct file mode.");
 
-		if (arg == "-") {
+		if (arg == "-") {		  
 			filename = "<stdout>";
+			filenames.push_back(filename);
```

Comment:
Why is `"<stdout>"` pushed here? You're ignoring it above and there are no other uses for `filenames`